src/v1/function-builder.ts

```diff
@@ -274,7 +274,7 @@ export function region(
  *
  * Value must not be null.
  */
-export function runWith(runtimeOptions: RuntimeOptions): FunctionBuilder {
+export function runWith(runtimeOptions: DeploymentOptions): FunctionBuilder {
```

Comment:
hmm this would mean things like `region` would also be suggested in `runWith` which shouldn't be.

I wonder I made the mistake of setting `preserveExternalChanges` Deployment Option instead of Runtime Option. It doesn't sound right that `preserveExternalChanges` is a runtime option, but then that's the only way I can abuse `runWith` tag 😢

Reply:
Hmm yeah I figured this was a bug since I couldn't figure out a way to use `preserveExternalChanges` without making this change. Also, the v1 docs show `preserveExternalChanges` in the runtime options section - https://firebase.google.com/docs/functions/manage-functions#set_runtime_options

Should we just dump `preserveExternalChanges` into `RuntimeOptions` instead? That way we wouldn't have to fix any docs

Reply:
let's dump it in RuntimeOptions. I think runWith is already a dumping of options.